src/NHibernate/AdoNet/ConnectionManager.cs

```diff
@@ -63,6 +68,16 @@ public interface Callback
 			ownConnection = suppliedConnection == null;
 		}
 
+		public void AddSessionSharingManager(ISessionImplementor session)
+		{
+			_sessionsSharingManager.Add(session);
+		}
+
+		public void RemoveSessionSharingManager(ISessionImplementor session)
+		{
+			_sessionsSharingManager.Remove(session);
```

Comment:
`LinkedList` is O(n) as `List` since it still does involve a non indexed search. It just saves on the list rewriting.